ocaml/typing/solver.ml

```diff
@@ -521,24 +526,22 @@ module Solver_mono (C : Lattices_mono) = struct
     mupper obj mv
 
   let zap_to_ceil : type a l. a C.obj -> (a, l * allowed) mode -> a =
-   fun obj -> function
+   fun obj m ->
+    match m with
     | Amode m -> m
     | Amodevar mv -> zap_to_ceil_morphvar obj mv
     | Amodemeet (a, mvs) ->
       List.fold_left
         (fun acc mv -> C.meet obj acc (zap_to_ceil_morphvar obj mv))
         a mvs
 
+  let cons_dedup obj x xs = if exists obj x xs then xs else x :: xs
+
   (* Similar to [List.append] but dedup the result (assuming both inputs are
```

Comment:
This also reverses the second list, putting it before the first. That's fine -- you don't care about order here -- but this is definitely not the same as `List.append`.

Reply:
Good point - `rev_append` sounds like a better name (there is already `List.rev_append`).